File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.py
class Solution:
    def bestClosingTime (self, customers : str) -> int:
        bestTime = 0; mmin = 0; pre = 0
        for i in range (len (customers)):
            pre += -1 if customers [i] == 'Y' else 1
            if pre < mmin:
                bestTime = i + 1
                mmin = pre
        return bestTime

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.java
class Solution {
    public int bestClosingTime (String customers) {
        int bestTime = 0, min = 0, pre = 0;
        for (int i = 0; i < customers.length (); i++) {
            pre += (customers.charAt (i) == 'Y') ? -1 : 1;
            if (pre < min) {
                bestTime = i + 1;
                min = pre;
            }
        }
        return bestTime;
    }
}